auth/server/src/main/java/com/seagates3/policy/PolicyUtil.java

```diff
@@ -65,7 +65,7 @@ static String getBucketFromUri(String uri) {
       return null;
     }
     String[] arr = uri.split("/");
-    if (arr[1] != null && !arr[1].isEmpty()) return arr[1];
+    if (arr.length != 0 && arr[1] != null && !arr[1].isEmpty()) return arr[1];

```

Comment:
I think the first check should be "arr.length > 0"
With current check the statement will still remain prone to error - ArrayOutOfBoundException if array has length = 1.
This may not occur in our case, but the code should not have any assumptions.

Reply:
No. It will not fail.
arr.length!=0 means arr.length >=1 
so it will never fail.

Reply:
![image](https://user-images.githubusercontent.com/45682869/110067528-f0e5ea80-7d99-11eb-9391-6b3c1718c147.png)

Reply:
This is what will occur if array has length 1.
As I said I am not sure if array of length 1 would be received, but if our assumption is arr.length can not be 0, then technically it can be anything. And our code should not make any such assumptions. Subtle errors should be handled.

Reply:
RIght. Making changes.

Reply:
Fixed